merkle/src/merkle.rs

```diff
@@ -346,11 +242,21 @@ impl<E: Element> ops::Deref for DiskStore<E> {
 
 impl<E: Element> Store<E> for DiskStore<E> {
     #[allow(unsafe_code)]
-    fn new(size: usize) -> Self {
+    fn new_with_path(size: usize, path: Option<&Path>) -> Self {
```

Comment:
A thought:

You might consider modifying `new_with_path` to accept a `File` instead of `Option<&Path>` because:

1. It prevents the caller from needing to try to figure out what will happen if it sends `path=None`
1. It means that the ugly business of handling errors at the file `open` call is handled by the caller

Reply:
@laser Sounds good, will change. Could you update the original issue in `rust-fil-proofs` to reflect that please and start shaping what would this API would look like?

Reply:
Updated to `Option<File>`.